src/pilot/config_idrac.py

```diff
@@ -208,6 +212,172 @@ def configure_bios_nics_boot_settings(drac_client, ip_service_tag, pxe_nic_id):
+    :raises: DRACOperationFailed on error reported back by the iDRAC
+             interface
+    """
+    return set_nic_setting(nic_id, 'LegacyBootProto', value, drac_client)
+
+
+def set_nic_legacy_boot_protocol_none(nic_id, drac_client):
+    """Set the legacy, non-UEFI, boot protocol of a NIC to NONE.
+
+    If successful, the pending value of the NIC's legacy boot
+    protocol attribute is set. For the new value to be applied, a
+    configuration job must be created and the node must be rebooted.
+
+    :param nic_id: id of the network interface controller (NIC)
+    :param drac_client: drac_client from python-dracclient
+    :returns: dictionary containing a 'commit_required' key with a
+              boolean value indicating whether a configuration job
+              must be created for the new legacy boot protocol
+              setting to be applied
+    :raises: WSManRequestFailure on request failures
+    :raises: WSManInvalidResponse when receiving invalid response
+    :raises: DRACOperationFailed on error reported back by the iDRAC
+             interface
+    """
+    return set_nic_legacy_boot_protocol(nic_id, 'NONE', drac_client)
+
+
+def set_nic_legacy_boot_protocol_pxe(nic_id, drac_client):
+    """Set the legacy, non-UEFI, boot protocol of a NIC to PXE.
+
+    If successful, the pending value of the NIC's legacy boot
+    protocol attribute is set. For the new value to be applied, a
+    configuration job must be created and the node must be rebooted.
+
+    :param nic_id: id of the network interface controller (NIC)
+    :param drac_client: drac_client from python-dracclient
+    :returns: dictionary containing a 'commit_required' key with a
+              boolean value indicating whether a configuration job
+              must be created for the new legacy boot protocol
+              setting to be applied
+    :raises: WSManRequestFailure on request failures
+    :raises: WSManInvalidResponse when receiving invalid response
+    :raises: DRACOperationFailed on error reported back by the iDRAC
+             interface
+    """
+    return set_nic_legacy_boot_protocol(nic_id, 'PXE', drac_client)
+
+
+def get_nic_setting(nic_id, attribute_name, drac_client):
+    """Obtain a setting of a NIC.
+
+    :param nic_id: id of the network interface controller (NIC)
+    :param attribute_name: name of the setting
+    :param drac_client: drac_client from python-dracclient
+    :returns: value of the attribute on successful query, None
+              otherwise
+    :raises: WSManRequestFailure on request failures
+    :raises: WSManInvalidResponse when receiving invalid response
+    :raises: DRACOperationFailed on error reported back by the iDRAC
+             interface
+    :raises: NotFound when no settings for NIC found
+    """
+    settings = drac_client.list_nic_settings(nic_id)
+
+    # Were no settings found?
+    if not settings:
+        raise exceptions.NotFound(
```

Comment:
Our long term objective is to completely eliminate discover_nodes/dracclient.  I took a look and only nic.py used this exception, and it is the only exception declared in discover_nodes/dracclient/exceptions.  In the case above, if no settings were found, then the nic_id passed in must not have been found, which means that the nic_id was invalid.  As a result, let's change the above to raise a dracclient.exceptions.InvalidParameterValue exception and change all of the references to NotFound to InvalidParameterValue.  Finally, delete the discover_nodes/dracclient/exceptions.py file.  This will get us one step closer to being able to delete discover_nodes/dracclient completely.